pkg/services/ngalert/notifier/channels/sensugo.go

```diff
@@ -0,0 +1,191 @@
+func NewSensuGoNotifier(model *models.AlertNotification, t *template.Template) (*SensuGoNotifier, error) {
+	if model.Settings == nil {
+		return nil, alerting.ValidationError{Reason: "No settings supplied"}
+	}
+
+	url := model.Settings.Get("url").MustString()
+	if url == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find URL property in settings"}
+	}
+
+	apikey := model.DecryptedValue("apikey", model.Settings.Get("apikey").MustString())
+	if apikey == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find the API key property in settings"}
+	}
+
+	return &SensuGoNotifier{
+		NotifierBase: old_notifiers.NewNotifierBase(model),
+		URL:          url,
+		Entity:       model.Settings.Get("entity").MustString(),
+		Check:        model.Settings.Get("check").MustString(),
+		Namespace:    model.Settings.Get("namespace").MustString(),
+		Handler:      model.Settings.Get("handler").MustString(),
+		APIKey:       apikey,
+		Message:      model.Settings.Get("message").MustString(`{{ template "default.message" .}}`),
+		log:          log.New("alerting.notifier.slack"),
+		tmpl:         t,
+	}, nil
+}
+
+// Notify sends an alert notification to Sensu Go
+func (sn *SensuGoNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	sn.log.Debug("Sending Sensu Go result")
+
+	data := notify.GetTemplateData(ctx, sn.tmpl, as, gokit_log.NewNopLogger())
+	var tmplErr error
+	tmpl := notify.TmplText(sn.tmpl, data, &tmplErr)
+
+	ruleUIDs := make(map[string]struct{})
+	ruleNames := make(map[string]struct{})
+	for _, a := range as {
+		v, ok := a.Labels[ngmodels.UIDLabel]
+		if ok {
+			_, exists := ruleUIDs[string(v)]
+			if !exists {
+				ruleUIDs[string(v)] = struct{}{}
+			}
+		}
+		v, ok = a.Labels[model.AlertNameLabel]
+		if ok {
+			_, exists := ruleNames[string(v)]
+			if !exists {
+				ruleNames[string(v)] = struct{}{}
+			}
+		}
+	}
+
+	var buffer bytes.Buffer
+	for k := range ruleUIDs {
+		buffer.WriteString(fmt.Sprintf("%s,", k))
+	}
+	strRuleUIDs := strings.TrimRight(buffer.String(), ",")
+
+	buffer.Reset()
+	for k := range ruleNames {
+		buffer.WriteString(fmt.Sprintf("%s,", k))
+	}
+	strRuleNames := strings.TrimRight(buffer.String(), ",")
+
+	// Sensu Go alerts require an entity and a check. We set it to the user-specified
+	// value (optional), else we fallback and use the grafana rule anme  and ruleID.
+	entity := sn.Entity
+	if entity == "" {
+		entity = "default"
+	}
+
+	check := sn.Check
+	if check == "" {
+		check = "default"
+	}
+
+	alerts := types.Alerts(as...)
+	status := 0
+	if alerts.Status() == model.AlertFiring {
+		// TODO figure out about NoData old state (we used to send status 1 in that case)
+		status = 2
+	}
+
+	namespace := sn.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	var handlers []string
+	if sn.Handler != "" {
+		handlers = []string{sn.Handler}
+	}
+
+	ruleURL := path.Join(sn.tmpl.ExternalURL.String(), "/alerting/list")
```

Comment:
This works for me
	ruleURL := sn.tmpl.ExternalURL
	ruleURL.Path = path.Join(ruleURL.Path, "/alerting/list")